arbos/storage/storage.go

```diff
@@ -442,12 +442,28 @@ func (sto *Storage) OpenStorageBackedBigInt(offset uint64) StorageBackedBigInt {
 }
 
 func (sbbi *StorageBackedBigInt) Get() (*big.Int, error) {
-	value, err := sbbi.StorageSlot.Get()
-	return value.Big(), err
+	asHash, err := sbbi.StorageSlot.Get()
+	asBig := new(big.Int).SetBytes(asHash[1:])
+	if asHash[0] != 0 {
+		asBig = new(big.Int).Neg(asBig)
+	}
+	return asBig, err
 }
 
 func (sbbi *StorageBackedBigInt) Set(val *big.Int) error {
-	return sbbi.StorageSlot.Set(common.BigToHash(val))
+	asBytes := val.Bytes()
+	if len(asBytes) > 31 {
+		panic("overflow in StorageBackedBigInt.Set")
+	}
+	asHash := common.BytesToHash(asBytes)
+	if val.Sign() < 0 {
+		asHash[0] = 1
```

Comment:
We should consider defining this as two's complement, as Solidity does

Reply:
Specifically, I think we can use `common.BigToHash(common.U256(val))` to encode this and `common.S256(val.Hash())` to decode this